src/twisted/names/client.py

```diff
@@ -439,6 +451,43 @@ def _cbLookupZone(self, result, connector):
 
 
 
+class IPv6Resolver(Resolver):
+    """
+    Selects only IPv6 resolv.conf entries.
+    """
+    def maybeAddNameserver(self, nameserver, servers):
+        resolver = (nativeString(nameserver), dns.PORT)
+        if b':' in nameserver:
```

Comment:
This is sort of an ugly way to check whether it's a v4 or v6 address, but it probably doesn't matter since I don't know of a single system that lets you specify an alternate port in `resolv.conf`.  The "nicest" way IMO would be to use the `ipaddress` package which is in the stdlib since 3.3, but requires an additional external dependency for py2, so I guess that's probably out.